### ISSUE TITLE
Build sometimes breaks failing to locate CUDA libraries

### ISSUE THREAD

**Issue by eyalroz** [eyalroz/cuda-api-wrappers#12]
I recently tried to build the wrapper use examples after switching from GNU/Linux Kubuntu 16.04 to Mint 18.1 (on the same machine). Strangely enough, I encountered a linking issue. The command:

```
/usr/bin/c++   -Wall -std=c++11 -g -g   CMakeFiles/device_management.dir/examples/by_runtime_api_module/device_management.cpp.o -o examples/bin/device_management -rdynamic lib/libcuda-api-wrappers.a -Wl,-Bstatic -lcudart_static -Wl,-Bdynamic -lpthread -ldl -lrt
```
fails to find the CUDA runtime library, and I get:
```
CMakeFiles/device_management.dir/examples/by_runtime_api_module/device_management.cpp.o: In function `cuda::device::peer_to_peer::get_attribute(cudaDeviceP2PAttr, int, int)':
/home/eyalroz/src/mine/cuda-api-wrappers/src/cuda/api/device.hpp:38: undefined reference to `cudaDeviceGetP2PAttribute'
collect2: error: ld returned 1 exit status
```
but if I add `-L/usr/local/cuda/lib64` it builds fine. This doesn't happen on bricks02
...
Fix this.

**Comment by eyalroz** [comment]
**Solution:** as you can see in [this StackOverflow question](http://stackoverflow.com/q/41416491/1593077), I need to explicitly have CMake add the appropriate `-L/path/to/cuda/libs` switch. Let's do it.

**Comment by eyalroz** [comment]
This seems to have been fixed by commit b42daa99d118ca49b90c5d177ae10d1c87176ce5 .
